packages/commonwealth/server/controllers/server_proposal_methods/compound/proposals.ts

```diff
@@ -166,6 +175,59 @@ async function getProposalDataSequentially(
   return [proposalCreatedEvents, proposals, proposalStates];
 }
 
+async function getProposalDataFieldsSequentially(
```

Comment:
We should convert this into a more functional style. Something like the following (pseudocode):
``` typescript
const proposalCreatedEvents = await getProposalCreatedEvents(contract);

const proposalPromises = proposalCreatedEvents.map(async (propCreatedEvent) => {
  const [
    proposer,
    eta,
    startBlock,
    endBlock,
    { forVotes, againstVotes, abstainVotes },
    state,
  ] = await Promise.all([
    contract.proposalProposer(propCreatedEvent.id),
    contract.proposalEta(propCreatedEvent.id),
    contract.proposalSnapshot(propCreatedEvent.id),
    contract.proposalDeadline(propCreatedEvent.id),
    contract.proposalVotes(propCreatedEvent.id),
    contract.state(propCreatedEvent.id),
  ]);

  return {
    id: propCreatedEvent.id,
    proposer,
    eta,
    startBlock,
    endBlock,
    forVotes,
    againstVotes,
    abstainVotes,
    canceled: state === 7,
    executed: state === 2,
  };
});

const [proposals, proposalStates] = await Promise.all([
  Promise.all(proposalPromises),
  Promise.all(proposalCreatedEvents.map((event) => contract.state(event.id))),
]);

return [proposalCreatedEvents, proposals, proposalStates];
```

Reply:
This looks great, but imo we should then change to this pattern for all the getProposalData functions